tracer.go

```diff
@@ -24,35 +24,45 @@ import (
 	"net/http"
 )
 
-/**
-Will be injecting the trace id and span id  in the logger.
-*/
-func InjectTraceInformationInLogger() logginghttp.LoggerFunc {
+// Will be injecting the trace id and span id  in the logger.
+func InjectTraceInformationInLogger(config Config) logginghttp.LoggerFunc {
 	return func(kv []interface{}, request *http.Request) []interface{} {
 		traceId, spanId := ExtractTraceInformation(request.Context())
-		return append(kv, SpanIdHeader, spanId, TraceIdHeader, traceId)
+		spanIDHeaderName, traceIDHeaderName := ExtractSpanIDAndTraceIDHeaderName(config)
+		return append(kv, spanIDHeaderName, spanId, traceIDHeaderName, traceId)
 	}
 }
 
-/**
-	Will be extracting the traceId and spanId
-	if  span is not started in middleware then it will be returning noopSpan
-	which will result traceId[32 digits] and spanId[16 digits] as 0
-	i.e. 00000000000000000000000000000000 and 0000000000000000
-*/
+// Will be extracting the traceId and spanId  if  span is not started in middleware then it will be
+// returning noopSpan which will result traceId[32 digits] and spanId[16 digits] as 0
+// i.e. 00000000000000000000000000000000 and 0000000000000000
+
 func ExtractTraceInformation(ctx context.Context) (string, string) {
 	span := trace.SpanFromContext(ctx)
 	traceId := span.SpanContext().TraceID.String()
 	spanId := span.SpanContext().SpanID.String()
 	return traceId, spanId
 }
 
-/**
-	Will be injecting  traceParent and tracestate headers in carrier
-	from span which is available in context.
-*/
+//	Will be injecting  traceParent and tracestate headers in carrier
+//	from span which is available in context.
+
 func InjectTraceInformation(ctx context.Context, carrier propagation.TextMapCarrier) {
 	prop := propagation.TraceContext{}
 	prop.Inject(ctx, carrier)
 
 }
+
+// Will be responsible for  providing the  SpanIDHeaderName and TraceIdHeaderName
+// if they are provided will be used else defaults will be used.
```

Comment:
This comment doesn't follow Effective Go guidelines. Please see other comments for examples on how it can be fixed, or let me know if you want further guidance.

Reply:
Modified comments as suggested.